src/ModelContextProtocol/Server/McpServerToolCreateOptions.cs

```diff
@@ -124,6 +125,15 @@ public sealed class McpServerToolCreateOptions
     /// </remarks>
     public bool? ReadOnly { get; set; }
 
+    /// <summary>
+    /// Gets or sets whether the tool should report an output schema for structured content.
+    /// </summary>
+    /// <remarks>
+    /// When enabled, the tool will attempt to populate the <see cref="Tool.OutputSchema"/>
+    /// and provide structured content in the <see cref="CallToolResponse.StructuredContent"/> property.
+    /// </remarks>
+    public bool UseStructuredContent { get; set; }
```

Comment:
Note that I've left this opt-in for now. This is because of the breaking change potential (in the case of AOT it can prompt schema generation for types that we're not trying to serialize) and because the feature is still in draft.

Reply:
For example: we special case [tools that return `IEnumerable<string>`](https://github.com/modelcontextprotocol/csharp-sdk/blob/57458241b987b7c2889662111e23a610b300a696/src/ModelContextProtocol/Server/AIFunctionMcpServerTool.cs#L321-L324) mapping the output to a list of content types. Because of structured outputs, we're now additionally exercising the JSON contract for `IEnumerable<string>` which can result in failures if you're using the source generator. I actually had a number of unit tests fail on me when I had this turned on by default.